core/src/main/java/org/apache/calcite/rel/rules/PruneEmptyRules.java

```diff
@@ -480,9 +486,24 @@ public interface JoinLeftEmptyRuleConfig extends PruneEmptyRule.Config {
       return new PruneEmptyRule(this) {
         @Override public void onMatch(RelOptRuleCall call) {
           Join join = call.rel(0);
+          Values empty = call.rel(1);
+          RelNode right = call.rel(2);
+          RexBuilder rexBuilder = call.builder().getRexBuilder();
           if (join.getJoinType().generatesNullsOnLeft()) {
             // "select * from emp right join dept" is not necessarily empty if
```

Comment:
Probably this comment (and the analogous one in `JoinRightEmptyRuleConfig`) needs to be adjusted because of this change.

Reply:
Updated the comment.